modules/ossm-integrating-with-user-workload-monitoring.adoc

```diff
@@ -0,0 +1,165 @@
++
+.Custom network policy
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: user-workload-access
+  namespace: bookinfo
+spec:
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          network.openshift.io/policy-group: monitoring
+  podSelector: {}
+  policyTypes:
+  - Ingress
+----
++
+
+. Apply a telemetry object to enable traffic metrics in Istio proxies:
++
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: enable-prometheus-metrics
+  namespace: istio-system
+spec:
+  metrics:
+  - providers:
+    - name: prometheus
+----
++
+[NOTE]
+====
+A telemetry object created in the control plane namespace applies to all workloads in a mesh. Set `spec.selector.matchLabels` to apply the telemetry object to specific namespaces or workloads.
+====
+
+. Apply a `ServiceMonitor` to monitor the Istio control plane:
++
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istiod-monitor
+  namespace: istio-system
+spec:
+  targetLabels:
+  - app
+  selector:
+    matchLabels:
+      istio: pilot
+  endpoints:
+  - port: http-monitoring
+    interval: 30s
+----
+
+. Apply a `PodMonitor` to collect metrics from Istio proxies:
++
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: istio-proxies-monitor
+  namespace: istio-system # <1>
+spec:
+  selector:
+    matchExpressions:
+    - key: istio-prometheus-ignore
+      operator: DoesNotExist
+  podMetricsEndpoints:
+  - path: /stats/prometheus
+    interval: 30s
+    relabelings:
+    - action: keep
+      sourceLabels: [__meta_kubernetes_pod_container_name]
+      regex: "istio-proxy"
+    - action: keep
+      sourceLabels: [__meta_kubernetes_pod_annotationpresent_prometheus_io_scrape]
+    - action: replace
+      regex: (\d+);(([A-Fa-f0-9]{1,4}::?){1,7}[A-Fa-f0-9]{1,4})
+      replacement: '[$2]:$1'
+      sourceLabels: [__meta_kubernetes_pod_annotation_prometheus_io_port, 
+      __meta_kubernetes_pod_ip]
+      targetLabel: __address__
+    - action: replace
+      regex: (\d+);((([0-9]+?)(\.|$)){4})
+      replacement: $2:$1
+      sourceLabels: [__meta_kubernetes_pod_annotation_prometheus_io_port, 
+      __meta_kubernetes_pod_ip]
+      targetLabel: __address__
+    - action: labeldrop
+      regex: "__meta_kubernetes_pod_label_(.+)"
+    - sourceLabels: [__meta_kubernetes_namespace]
+      action: replace
+      targetLabel: namespace
+    - sourceLabels: [__meta_kubernetes_pod_name]
+      action: replace
+      targetLabel: pod_name
+    - action: replace
+      replacement: "<SMCP_NAME>-<SMCP_NAMESPACE>" # <2>
+      targetLabel: mesh_id
+----
+<1> Since OpenShift Monitoring ignores the field `spec.namespaceSelector` in `ServiceMonitor` and `PodMonitor` objects, you need to apply the `PodMonitor` in all mesh namespaces, including the control plane namespace.
```

Comment:
```suggestion
<1> Since OpenShift Monitoring ignores the field `spec.namespaceSelector` in `ServiceMonitor` and `PodMonitor` objects, you need to apply the resource to the namespace it should apply to. You can apply it to the control plane namespace to apply it for all mesh namespaces.
```
Just elaborating on this to explain that you can restrict ServiceMonitors/PodMonitors to specific namespaces if desired.

Reply:
> You can apply it to the control plane namespace to apply it for all mesh namespaces.

This statement is not correct. OpenShift monitoring stack does not know anything about mesh namespaces. On the other hand, Istio does not handle Service/Pod monitors, so this semantic does not apply to monitoring resources.